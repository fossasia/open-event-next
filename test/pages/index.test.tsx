import React from 'react'
import { render, fireEvent } from '../testUtils'
import Index from '../../pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    interface Props {
      events: ServerData[]
      upcomingEvents: ServerData[]
      groups: ServerData[]
    }

    const events: ServerData[] = [
      {
        type: 'event',
        attributes: {
          name: 'Event One',
          startsAt: new Date('March 12, 2022 1:45:00'),
          endsAt: new Date('March 15, 2022, 1:44:00'),
          thumbnailImageUrl:
            'https://api.eventyay.com/static/media/events/2764/thumbnail/TXAvdWNJQV/86acb2e9-7034-40fd-96fc-7453d1b58059.jpg',
          timezone: 'Europe/Berlin',
          online: false,
        },
      },
      {
        type: 'event',
        attributes: {
          name: 'Event Two',
          startsAt: new Date('February 12, 2022 2:00:00'),
          endsAt: new Date('February 17, 2022, 1:59:00'),
          thumbnailImageUrl:
            'https://api.eventyay.com/static/media/events/2764/thumbnail/TXAvdWNJQV/86acb2e9-7034-40fd-96fc-7453d1b58059.jpg',
          timezone: 'Singapore',
          online: false,
        },
      },
    ]
    const upcomingEvents: ServerData[] = [...events]
    const groups: ServerData[] = [
      {
        type: 'group',
        attributes: {
          name: 'Group One',
          thumbnailImageUrl:
            'https://api.eventyay.com/static/media/events/40/thumbnail/V3FyRzk3WF/059a6ce5-b592-448d-ac00-ca7185be4adf.jpg',
          about: 'This is some random test string about group one',
          followerCount: 3,
        },
      },
      {
        type: 'group',
        attributes: {
          name: 'Group Two',
          thumbnailImageUrl:
            'https://api.eventyay.com/static/media/events/40/thumbnail/V3FyRzk3WF/059a6ce5-b592-448d-ac00-ca7185be4adf.jpg',
          about: 'This is some random test string about group two',
          followerCount: 3,
        },
      },
    ]
    const props: Props = { events, upcomingEvents, groups }
    const { asFragment } = render(
      <Index events={events} upcomingEvents={upcomingEvents} groups={groups} />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
