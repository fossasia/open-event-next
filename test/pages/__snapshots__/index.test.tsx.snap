// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p>
        This is line to test translation.
      </p>
      <button>
        English
      </button>
      <button>
        Hindi
      </button>
      <div>
        <p>
          LOCAL : Apr, 04, 2020 04:00 PM
        </p>
        <p>
          SGT : Apr, 05, 2020 12:00 AM
        </p>
        <select
          id="cars"
          name="timezone"
        >
          <option
            value="Asia/Calcutta"
          >
            Asia/Calcutta
          </option>
          <option
            value="Asia/Singapore"
          >
            Asia/Singapore
          </option>
        </select>
      </div>
      <p
        class="description"
      >
        Get started by editing 
        <code
          class="code"
        >
          pages/index.js
        </code>
      </p>
      <button>
        Test Button
      </button>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Testing Material UI Button
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              aria-label="Checkbox A"
              class="PrivateSwitchBase-input-4"
              data-indeterminate="false"
              type="checkbox"
              value="checkedA"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
      </div>
      <form>
        <fieldset
          class="MuiFormControl-root"
        >
          <legend
            class="MuiFormLabel-root"
          >
            Pop quiz: Can you see Button and checkbox above
          </legend>
          <div
            aria-label="quiz"
            class="MuiFormGroup-root"
            role="radiogroup"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-4"
                    name="quiz"
                    type="radio"
                    value="Yes"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-5"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Yes
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-4"
                    name="quiz"
                    type="radio"
                    value="No"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-5"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                No
              </span>
            </label>
          </div>
          <p
            class="MuiFormHelperText-root"
          >
            Choose Wisely
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Check Answer
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </fieldset>
      </form>
      <div
        class="grid"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h3>
            Documentation →
          </h3>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="card"
          href="https://nextjs.org/learn"
        >
          <h3>
            Learn →
          </h3>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3>
            Examples →
          </h3>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="card"
          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h3>
            Deploy →
          </h3>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="logo"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
